src/poisson_dev_fe.jl

```diff
@@ -571,6 +571,21 @@ low_level_manual_gradient_dv_array = lazy_map(Broadcasting(Operation(⋅)),inv_J
 
 # ## A low-level implementation of the residual integration and assembly
 
+# In the rest of the tutorial we aim to solve a Poisson equation with homogeneous source term, i.e., $f=0$ and non-homogeneous Dirichlet boundary conditions $u=g_0$ on $\Gamma_D$, with  $\Gamma_D$ being the whole boundary of the model. While the strong imposition of non-homogeneous Dirichlet boundary conditions in Gridap is done under the hood by modifying the global assembly process by subtracting the contributions of boundary conditions to the right hand side of the linear system, in this tutorial, for simplicity, we follow a different approach. This approach requires: (1) to be able to assemble the residual of the PDE for an arbitrary finite element function $\hat{u}_h$ (current section); (2) to be able to assemble the coefficient matrix of the finite element linear system (next section). We briefly outline this solution approach in the sequel.
+
+# The discretized variational form of the Poisson problem reads as:
+# >Find $u_h \in V_h^{\Gamma_D}=\{v_h \in V_h:v_h=g_0 \; on \; \Gamma_D\}$ such that
```

Comment:
There is a subtle, but important mathematical detail in here. At the continuous problem level, we can enforce `v=g_0`. However, this is not actually the case at the discrete problem level, as, generally,  `g_0` might not be in the FE space. What we do in practice is to enforce `v_h=\Pi_h(g_0)` where $\Pi_h$ denotes a suitable projection operator, for example, the interpolation operator. Can you modify the definition of the FE space to reflect this subtletie? Thanks!

Reply:
Yes, that's important to notice. I prefer to use $g_0^h$ to denote the discretized version of function $g_0$.

Reply:
ok, let us use $g_0^h$, and add a comment to briefly define how it is defined.